### PR TITLE
Fix DCA voters not checking module and parent update access

### PR DESCRIPTION
**Pull request by aschempp**
I wanted to implement new voters… but came across a lot of issues in the current ones 🙈.

1. Voters must also check if a user has access to the respective back end modules
2. A child table voter must make sure an `UpdateAction` does not modify the parent ID or changes it to an allowed parent ID

As soon as this and https://github.com/contao/contao/pull/6627 is merged I'll also need to refactor the `isGranted` check, but I'll keep this separate to be easier to review.

### CODE REVIEW

File: calendar-bundle/src/Security/Voter/CalendarAccessVoter.php [hunk 1]
```diff
@@ -33,6 +36,10 @@ protected function getTable(): string
 
     protected function isGranted(CreateAction|DeleteAction|ReadAction|UpdateAction $action): bool
     {
+        if (!$this->security->isGranted(ContaoCalendarPermissions::USER_CAN_ACCESS_MODULE)) {
```
Comment (leofeyer) [inline calendar-bundle/src/Security/Voter/CalendarAccessVoter.php:39]: Just out of curiosity, why did you add an extra constant here instead of using
`->isGranted(ContaoCorePermissions::USER_CAN_ACCESS_MODULE, 'calendar')`?

**Review by leofeyer — APPROVED** [review]

File: calendar-bundle/src/Security/Voter/CalendarAccessVoter.php (hunk 1 above)
Comment (aschempp) [reply calendar-bundle/src/Security/Voter/CalendarAccessVoter.php:39]: Just for convenience. Because I can never remember if its calendar, calendars, events etc 😂

**Comment by leofeyer** [comment]
Thank you @aschempp.
